File: 0151-reverse-words-in-a-string/0151-reverse-words-in-a-string.java
import java.util.*;

class Solution {
    public String reverseWords(String s) {
        StringBuilder sb = new StringBuilder();
        String[] words = s.trim().split("\\s+"); // Split by spaces and remove extra spaces
        
        for (int i = words.length - 1; i >= 0; i--) {
            sb.append(words[i]); // Add each word in reverse order
            if (i > 0) {
                sb.append(" "); // Add space between words
            }
        }

        return sb.toString();
    }
}

File: 0151-reverse-words-in-a-string/0151-reverse-words-in-a-string.cs
public class Solution {
    public string ReverseWords(string s) {
        string[] str = s.Split(' ', StringSplitOptions.RemoveEmptyEntries);

        StringBuilder sb = new StringBuilder();

        // Traverse words from the end to the start
        for (int i = str.Length - 1; i >= 0; i--) {
            sb.Append(str[i]);
            if (i > 0) sb.Append(' '); // avoid extra space at the end
        }

        return sb.ToString();
    }
}